File: api/src/main/java/com/diamond/workoutplanner/workoutplanexercise/dto/WorkoutPlanExerciseResponse.java
package com.diamond.workoutplanner.workoutplanexercise.dto;

import com.diamond.workoutplanner.exercise.MuscleGroup;

public record WorkoutPlanExerciseResponse(
        int id,
        int exerciseId,
        String exerciseName,
        MuscleGroup muscleGroup,
        String instructions,
        int targetSets,
        int targetReps
) {
}

File: frontend/src/types/WorkoutPlanExerciseResponse.ts
import type { MuscleGroup } from './MuscleGroup';

export interface WorkoutPlanExerciseResponse {
  id: number;
  exerciseId: number;
  exerciseName: string;
  muscleGroup: MuscleGroup;
  instructions: string;
  targetSets: number;
  targetReps: number;
}